dashboard/app/controllers/profanity_controller.rb

```diff
@@ -4,8 +4,18 @@ class ProfanityController < ApplicationController
   # POST /profanity/find
   # @param [String] params[:text] String to test
   # @param [String] params[:language] Language of the text to test
+  # @param [Boolean] params[:should_cache] Whether or not we should use the cache on this request
   # @returns [Array<String>|nil] Profane words within the given string
   def find
-    render json: ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
+    profanity_result = []
+    if params[:should_cache]
+      cache_key = "profanity/#{params[:language] || request.locale}/#{params[:text] || ''}"
+      profanity_result = Rails.cache.fetch(cache_key) do
+        ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
+      end
+    else
+      profanity_result = ProfanityFilter.find_potential_profanities(params[:text] || "", params[:language] || request.locale)
```

Comment:
I guess this question applies to existing code too, but is there any point calling `find_potential_profanities` if `params[:text]` is `nil`?

Reply:
not really, I'll add a check for it